src/details/ArborX_MinimumSpanningTree.hpp

```diff
@@ -0,0 +1,504 @@
+/****************************************************************************
+ * Copyright (c) 2017-2021 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef ARBORX_MINIMUM_SPANNING_TREE_HPP
+#define ARBORX_MINIMUM_SPANNING_TREE_HPP
+
+#include <ArborX_AccessTraits.hpp>
+#include <ArborX_DetailsKokkosExtArithmeticTraits.hpp>
+#include <ArborX_DetailsKokkosExtMinMaxOperations.hpp>
+#include <ArborX_DetailsMutualReachabilityDistance.hpp>
+#include <ArborX_DetailsTreeNodeLabeling.hpp>
+#include <ArborX_DetailsUtils.hpp>
+#include <ArborX_LinearBVH.hpp>
+
+#include <Kokkos_Core.hpp>
+
+namespace ArborX
+{
+namespace Details
+{
+
+struct WeightedEdge
+{
+  int source;
+  int target;
+  float weight;
+
+private:
+  // performs lexicographical comparison by comparing first the weights and then
+  // the unordered pair of vertices
```

Comment:
I don't think this comment brings any clarity. I'd rather either remove it, or explicitly state how things are compared.